### PR TITLE
feat: Allow disabling backend notifications via env var

### PR DESCRIPTION
**Pull request by pgrzesik**
Addresses: https://github.com/serverless/serverless/issues/9303

### CODE REVIEW

File: process-backend-notification-request.js [hunk 1]
```diff
@@ -49,6 +78,17 @@ module.exports = (notifications) => {
           );
           return false;
         }
+
+        if (
+          notificationsMode !== NOTIFICATIONS_MODE_ON &&
+          !(
+            notificationsMode === NOTIFICATIONS_MODE_ONLY_OUTDATED_VERSION &&
+            OUTDATED_VERSION_NOTIFICATION_CODES.has(notification.code)
+          )
```
Comment (medikoo) [inline process-backend-notification-request.js:87]: Wouldn't it be more readable to construct it as:

```javascript
if (notificationsMode === NOTIFICATIONS_MODE_OFF) return false;
if (notificationsMode === NOTIFICATIONS_MODE_ONLY_OUTDATED_VERSION && !OUTDATED_VERSION_NOTIFICATION_CODES.has(notification.code)) {
  return false;
}
```

At least I'm looking at current condition, and after 30 seconds of mind bending, I'm still not sure if there's no issue with it :)

**Review by medikoo — COMMENTED** [review]
@pgrzesik looks good. I have just proposal to make one condition more readable

File: process-backend-notification-request.js (hunk 1 above)
Comment (pgrzesik) [reply process-backend-notification-request.js]: good suggestions, your proposal definitely is much more readable :+1:

**Review by medikoo — APPROVED** [review]
Looks great 👍

**Pull request merged**